src/main/java/io/wisetime/connector/jira/JiraConnector.java

```diff
@@ -51,6 +54,11 @@
 public class JiraConnector implements WiseTimeConnector {
 
   private static final Logger log = LoggerFactory.getLogger(WiseTimeConnector.class);
+
+  public static final Pattern EMAIL_REGEX = Pattern.compile(
+      "^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,6}$", Pattern.CASE_INSENSITIVE
```

Comment:
Simple email regex just to check if external id looks like an email (https://stackoverflow.com/questions/8204680/java-regex-email)

There's is an existing Apache library for email validation but its implementation is not a silver bullet (see docs http://commons.apache.org/proper/commons-validator/apidocs/org/apache/commons/validator/routines/EmailValidator.html)

Reply:
We had an email validation problem in the front-end recently, because the regex was too restrictive. This one is too - `.[A-Z]{2,6}` makes the domain only up to 6 characters, since the latest conventions allow it to be fully blown host name. I would suggest a less restrictive one we ended up using in the [frontend app](https://stash.practiceinsight.io/stash/projects/WT/repos/wt-customer-frontend/browse/src/main/common/formatters.js#6). It makes [all these tests](https://stash.practiceinsight.io/stash/projects/WT/repos/wt-customer-frontend/browse/src/tests/unit/utils/formatters-isValidEmail.test.js) not annotated with `xit` or `xdescribe` pass.

An even simpler and often suggested solution that will be even less restrictive and correct is to check if the string contains `@` and the domain part (after `@`) has at least one `.` 

An almost comprehensive description of the valid emails is in the doc comment in the [test file](https://stash.practiceinsight.io/stash/projects/WT/repos/wt-customer-frontend/browse/src/tests/unit/utils/formatters-isValidEmail.test.js).

Reply:
Ok, I agree. I'll make it less restrictive